### DISCUSSION TITLE
component.type.ts

### DISCUSSION THREAD

**Discussion by Noelsky-code** [AjouCapstone-Sw/client#11] [General]
이번 기회에 coponent.type.ts 에 대한 컨벤션을 명확히 하고 가면 좋을 것 같아서 discussion 오픈했습니다 

https://github.com/AjouCapstone-Sw/client/pull/9#discussion_r996436831

<img width="454" alt="스크린샷 2022-10-17 오후 12 14 35" src="https://user-images.githubusercontent.com/66724340/196081590-73dc8d7b-cccc-498e-9c16-dd7c4379b3da.png">

혹시 type.ts 파일로 type을 분리하는데에 생각하신 기준이 있을까요? 
이미 작성된 Button 컴포넌트의 type 도 분리되어있지 않습니다. 

코드 컨벤션에도 타입 분리에 대한 내용이 없어 논의할만한 주제인 것 같습니다. 

이 점에 대해서 저는 1개의 컴포넌트 파일에 1개의 타입만 정의되어있다면, 그 타입이 컴포넌트와 1:1 매핑관계로 강하게 의존되어있다고 생각합니다.

그래서 해당 타입이 다른 컴포넌트에 의해 외부에 참조되지 않는다면 타입을 별도의 파일로 분리하는게 
타입 수정, 타입 생성, 타입 확인 등 모든 작업에 불필요한 depth를 1단계 추가하면서 생산성을 저해한다고 생각합니다.

**Comment by jin-Pro** [comment]
1개의 타입만 사용된다면 타입을 별도의 파일로 분리하게된다면,
파일의 갯수도 많아 질 것이고,
말씀 주신대로 작업에 불필요한 depth가 1단계 늘어날 것이라 생각하는데요.

컨벤션을 작성하는 이유가 통일성을 유지하기 위함이라 생각해서
저의 생각은 파일 분리가 맞다고 생각이 듭니다.
